app/components/Layout.js

```diff
@@ -32,126 +29,114 @@ import {
 } from "utils/routeHelpers";
 
 type Props = {
-  documents: DocumentsStore,
   children?: ?React.Node,
   actions?: ?React.Node,
   title?: ?React.Node,
-  auth: AuthStore,
-  ui: UiStore,
   notifications?: React.Node,
-  i18n: Object,
-  t: TFunction,
 };
 
-@observer
-class Layout extends React.Component<Props> {
-  scrollable: ?HTMLDivElement;
-  @observable redirectTo: ?string;
-  @observable keyboardShortcutsOpen: boolean = false;
+function Layout({ children, actions, title, notifications }: Props) {
+  const [redirectTo, setRedirectTo] = React.useState(undefined);
+  const [keyboardShortcutsOpen, setKeyboardShortcutsOpen] = React.useState(
+    false
+  );
 
-  componentDidUpdate() {
-    if (this.redirectTo) {
-      this.redirectTo = undefined;
+  React.useEffect(() => {
+    if (redirectTo) {
+      setRedirectTo(undefined);
     }
-  }
+  }, [redirectTo]);
 
-  @keydown(`${meta}+.`)
-  handleToggleSidebar() {
-    this.props.ui.toggleCollapsedSidebar();
-  }
+  const { auth, ui } = useStores();
+  const { t } = useTranslation();
 
-  @keydown("shift+/")
-  handleOpenKeyboardShortcuts() {
-    this.keyboardShortcutsOpen = true;
-  }
+  useHotkeys(`${meta}+.`, () => {
+    ui.toggleCollapsedSidebar();
+  });
 
-  handleCloseKeyboardShortcuts = () => {
-    this.keyboardShortcutsOpen = false;
-  };
+  useHotkeys("shift+/", () => {
+    setKeyboardShortcutsOpen(true);
+  });
 
-  @keydown(["t", "/", `${meta}+k`])
-  goToSearch(ev: SyntheticEvent<>) {
+  const handleCloseKeyboardShortcuts = React.useCallback(() => {
+    setKeyboardShortcutsOpen(false);
+  }, []);
+
+  useHotkeys("t", "/", `${meta}+k`, (ev: SyntheticEvent<>) => {
     ev.preventDefault();
     ev.stopPropagation();
     this.redirectTo = searchUrl();
```

Comment:
Sorry, this `redirectTo` approach is a bit of an old pattern that we need to remove, can use a standard `history.push(route)` like you've done in `InputSearch` already for these 😄